bkpmgmt/history.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import logging
+import sqlite3
+
+
+logger = logging.getLogger(__name__)
+
+
+class History:
+    """Store and show history of commands done with the bkpmgmt tool.
+
+    :ivar string dbpath: Path to the history database.
+    """
+
+    def __init__(self, dbpath):
+        self._path = None
+        self._conn = None
+
+        self._path = dbpath
+        self._conn = sqlite3.connect(self._path)
+        logger.debug(
+            "Opened database {0} successfully".format(self._path)
+        )
+        self._conn.execute(
+            '''CREATE TABLE IF NOT EXISTS history
+                (
+                    id          INTEGER PRIMARY KEY AUTOINCREMENT   NOT NULL,
+                    datetime    REAL                                NOT NULL,
+                    command     TEXT                                NOT NULL,
+                    customer    TEXT                                NOT NULL,
+                    vault       TEXT,
+                    size        TEXT
+                )
+            ;'''
+        )
+        self._conn.close()
+
+    def add(self, customer, command, vault=None, size=None):
+        """Add an entry to the history.
+
+        :param string customer: Customer name.
+        :param string command:  Used subcommand, e.g. create or resize.
+        :param string vault:    Vault name or server hostname.
+        :param string size:     Quota size.
+
+        :returns: True
+        :rtype: bool
+
+        :raises sqlite3.OperationalError: if something with the database goes
+        wrong.
+        """
+        self._conn = sqlite3.connect(self._path)
```

Comment:
Would it not be simpler to open up the connection in the constructor and leave it open?